### PR TITLE
Add Sphinx documentation with nbsphinx

### PR DESCRIPTION
**Pull request by mrocklin**
This adds basic sphinx quickstart material and then uses nbsphinx to construct
rst pages.

This also slightly modifies many of the notebooks to make them slightly more
uniform in the documentation.

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @TomAugspurger

**Comment by mrocklin** [comment]
image

image

**Comment by mrocklin** [comment]
This is with https://github.com/dask/dask-sphinx-theme/pull/8

File: index.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
+Dask Examples
+=============
+
+These examples show how to use Dask in a variety of situations.
+
+First, there are some high level examples about various Dask APIs like arrays,
+dataframes, and futures, then there are more in-depth examples about
+particularly features or use cases.
```
Comment (TomAugspurger) [inline index.rst]: typo: particular

File: conf.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+# for a list of supported languages.
+#
+# This is also used if you do content translation via gettext catalogs.
+# Usually you set "language" from the command line for these cases.
+language = None
+
+# List of patterns, relative to source directory, that match files and
+# directories to ignore when looking for source files.
+# This pattern also affects html_static_path and html_extra_path .
+exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']
```
Comment (TomAugspurger) [inline conf.py]: May want to add `"**.ipynb_checkpoints"`.

**Comment by mrocklin** [comment]
> LGTM. Does the default cell timeout of 60s work for all the notebooks?

Things seem ok.  The full process runs in three minutes on my machine.  I imagine that things might operate differently on CI systems though.

**Comment by TomAugspurger** [comment]
+1

File: index.rst [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Basic Examples:
+
+   array
+   dataframe
+   delayed
+   futures
+   machine-learning
```
Comment (guillaumeeb) [inline index.rst]: What about embarrassingly-parallel example 😢 

Joking aside, I can understand it's difficult to put it somewhere in this page.

**Comment by TomAugspurger** [comment]
A bit too long on the tpot example. We can bump the timeout, or change the example to have fewer generations / fewer individuals.

**Comment by mrocklin** [comment]
I'm also experimenting with this on my master branch with rtd.  Hence being
a bit quiet here.

**Comment by mrocklin** [comment]
OK, so I'm having difficulty getting this to run on RTD.  Odd things are happening with dask.distributed: https://readthedocs.org/projects/dask-examples/builds/7787173/

Putting that up here in case anyone has any thoughts?

@TomAugspurger do you have any interest in setting up doctr here instead to see if that is smoother?

**Comment by TomAugspurger** [comment]
Sure I can take care of doctr

**Pull request merged**